gems/aws-sdk-s3/lib/aws-sdk-s3/file_downloader.rb

```diff
@@ -31,7 +31,13 @@ def download(destination, options = {})
           key: options[:key],
         }
         @params[:version_id] = options[:version_id] if options[:version_id]
-        @params[:checksum_mode] = options[:checksum_mode] || 'ENABLED'
+
+        # checksum_mode only supports the value "ENABLED"
+        # falsey values (false/nil) should be considered disabled and
+        # the api parameter should be unset.
+        if options.fetch(:checksum_mode, 'ENABLED')
```

Comment:
I think you can assign this to checksum_mode in the if statement to avoid fetching twice.